aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,500 @@
+    }
+}
+
+impl<'a, S> TStateView for SimulationStateView<'a, S>
+where
+    S: StateView,
+{
+    type Key = StateKey;
+
+    fn get_state_value(&self, state_key: &Self::Key) -> StateStoreResult<Option<StateValue>> {
+        if let Some(res) = self.states.lock().unwrap().get(state_key) {
+            return Ok(res.clone());
+        }
+        self.remote.get_state_value(state_key)
+    }
+
+    fn get_usage(&self) -> StateStoreResult<StateStorageUsage> {
+        panic!("not supported")
+    }
+
+    fn as_in_memory_state_view(&self) -> InMemoryStateView {
+        panic!("not supported")
+    }
+}
+
+/***************************************************************************************************
+ * Move Patches
+ *
+ **************************************************************************************************/
+static MODULE_ID_CREATE_SIGNER: Lazy<ModuleId> = Lazy::new(|| {
+    ModuleId::new(
+        AccountAddress::ONE,
+        Identifier::new("create_signer").unwrap(),
+    )
+});
+
+static MODULE_ID_APTOS_GOVERNANCE: Lazy<ModuleId> = Lazy::new(|| {
+    ModuleId::new(
+        AccountAddress::ONE,
+        Identifier::new("aptos_governance").unwrap(),
+    )
+});
+
+/// Makes `aptos_framework::create_signer::create_signer` a public function so that
+/// it can be used freely in the simulation.
+fn patch_create_signer(state_view: &SimulationStateView<impl StateView>) -> Result<()> {
+    let resolver = state_view.as_move_resolver();
+    let blob = resolver
+        .get_module(&MODULE_ID_CREATE_SIGNER)?
+        .ok_or_else(|| anyhow!("module create_signer does not exist"))?;
+
+    let mut m = CompiledModule::deserialize(&blob)?;
+    let func_def = find_function_def_by_name(&mut m, IdentStr::new("create_signer").unwrap())
+        .ok_or_else(|| anyhow!("failed to locate `fun create_signer`"))?;
+    func_def.visibility = Visibility::Public;
+
+    move_bytecode_verifier::verify_module(&m)?;
+
+    let mut blob = vec![];
+    m.serialize(&mut blob)?;
+
+    state_view.set_state_value(
+        StateKey::module_id(&MODULE_ID_CREATE_SIGNER),
+        StateValue::new_legacy(blob.into()),
+    );
+
+    Ok(())
+}
+
+/// Patches `aptos_framework::aptos_governance::resolve_multi_step_proposal` so that
+/// it returns the requested signer directly, skipping the governance process altogether.
+fn patch_resolve_multistep_proposal(
+    state_view: &SimulationStateView<impl StateView>,
+) -> Result<()> {
+    use Bytecode::*;
+
+    let resolver = state_view.as_move_resolver();
+    let blob = resolver
+        .get_module(&MODULE_ID_APTOS_GOVERNANCE)?
+        .ok_or_else(|| anyhow!("module create_signer does not exist"))?;
+
+    let mut m = CompiledModule::deserialize(&blob)?;
+    let create_signer_handle_idx = get_or_add_simple_function_handle(
+        &mut m,
+        MODULE_ID_CREATE_SIGNER.address,
+        MODULE_ID_CREATE_SIGNER.name.clone(),
+        Identifier::new("create_signer").unwrap(),
```

Comment:
nit: we can also have a const for `create_signer` function name? We already do this for module id.